docs/src/main/asciidoc/telemetry-opentracing-to-otel-tutorial.adoc

```diff
@@ -0,0 +1,504 @@
+|===
+
+Once the dependencies have been updated, the above class will break the build because the quickstart project is now running with OpenTelemetry. Errors like this will show up in the logs:
+
+[source,bash]
+----
+2023-10-27 16:11:12,454 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
+        [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.opentracing.Tracer and qualifiers [@Default]
+...
+----
+
+The new OpenTelemetry API must be used instead. This is one way to migrate the code:
+
+=== GreetingsResource with OpenTelemetry manual instrumentation
+[source,java]
+----
+package org.acme;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import static io.opentelemetry.api.trace.StatusCode.*;
+
+@Path("/hello")
+@ApplicationScoped
+public class GreetingResource {
+
+    @Inject
+    io.opentelemetry.api.trace.Tracer otelTracer;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @WithSpan(value = "Not needed, will create a new span, child of the automatic JAX-RS span")
+    public String hello() {
+        // Add a tag to the active span
+        Span incomingSpan = Span.current();
+        incomingSpan.setAttribute(SemanticAttributes.CODE_NAMESPACE, "GreetingResource");
+
+        // Create a manual inner span
+        Span innerSpan = otelTracer.spanBuilder("Count response chars").startSpan();
+        try (Scope scope = innerSpan.makeCurrent()) {
+            final String response = "Hello from RESTEasy Reactive";
+            innerSpan.setAttribute("response-chars-count", response.length());
+            return response;
+        } catch (Exception e) {
+            innerSpan.setStatus(ERROR);
+            innerSpan.recordException(e);
+            throw e;
+        } finally {
+            innerSpan.end();
+        }
+    }
+}
+
+----
+Once you remove all the OpenTracing dependencies the code will build. Don't forget to double check if the traces contain the right spans. You can see them in the Jaeger UI: http://localhost:16686/.
+
+== The OpenTracing shim
+
+On this part we present the OpenTelemetry library that can smooth the transition by providing access to the legacy OpenTracing API. This can help with the migration of large applications with many manual instrumentation points.
```

Comment:
```suggestion
In this section, we present an OpenTelemetry library that can smooth the transition by providing access to the legacy OpenTracing API. This can help with the migration of large applications with many manual instrumentation points.
```